### PR TITLE
fix `ss-redir -v`.

### PR DESCRIPTION
**Pull request by kimw**
It's now output messages while redirecting.
This is useful at the time building up a router.

### CODE REVIEW

File: src/redir.c [hunk 1]
```diff
@@ -191,6 +191,26 @@ static void server_recv_cb(EV_P_ ev_io *w, int revents)
...
     }
 
+    if (verbose) {
+        uint16_t port = 0;
+        char ipstr[max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN)];
+        memset(&ipstr, 0, max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN));
+
+        if (AF_INET == server->destaddr.ss_family) {
+            struct sockaddr_in *sa = (struct sockaddr_in *)&(server->destaddr);
+            inet_ntop(AF_INET, &(sa->sin_addr), ipstr, INET_ADDRSTRLEN);
```
Comment (madeye) [inline src/redir.c]: Please use `dns_ntop` instead.

File: src/redir.c [hunk 2]
```diff
@@ -191,6 +191,26 @@ static void server_recv_cb(EV_P_ ev_io *w, int revents)
...
+            port = ntohs(sa->sin_port);
+        } else {
+            // TODO: The code below need to be test in IPv6 envirment, which I
+            //       don't have.
+            struct sockaddr_in6 *sa = (struct sockaddr_in6 *)&(server->destaddr);
+            inet_ntop(AF_INET6, &(sa->sin6_addr), ipstr, INET6_ADDRSTRLEN);
```
Comment (madeye) [inline src/redir.c]: Ditto.

File: src/redir.c [hunk 3]
```diff
@@ -191,6 +191,26 @@ static void server_recv_cb(EV_P_ ev_io *w, int revents)
         }
     }
 
+    if (verbose) {
+        uint16_t port = 0;
+        char ipstr[max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN)];
```
Comment (madeye) [inline src/redir.c]: Just use INET6_ADDRSTRLEN here.

Comment (kimw) [inline src/redir.c]: It's a safe code consideration.

Should I use MAX() instead of max()?

Comment (madeye) [inline src/redir.c]: It's safe to assume INET6_ADDRSTRLEN > INET_ADDRSTRLEN. You can remove the comparison here.

File: src/redir.c (hunk 2 above)
Comment (kimw) [inline src/redir.c]: Done.

File: src/redir.c (hunk 1 above)
Comment (kimw) [inline src/redir.c]: checked

File: src/redir.c (hunk 3 above)
Comment (kimw) [inline src/redir.c]: It's removed.

2 problems are all handled.

**Comment by madeye** [comment]
Merged, thanks!

**Pull request merged**